app/Filament/Resources/DocumentResource/RelationManagers/RevisionsRelationManager.php

```diff
@@ -0,0 +1,87 @@
+
+class RevisionsRelationManager extends RelationManager
+{
+    protected static string $relationship = 'revisions';
+
+    public function form(Form $form): Form
+    {
+        return $form
+            ->schema([
+                Forms\Components\DatePicker::make('date')
+                    ->label(__('Revision date'))
+                    ->format('Y-m-d')
+                    ->unique()
+                    ->default(Carbon::now())
+                    ->validationMessages([
+                        'unique' => __('A revision with this date already exists.'),
+                    ])
+                    ->columnSpan(2),
+                Forms\Components\FileUpload::make('file.en')
+                    ->label(__('File (English)'))
+                    ->requiredWithout('file.fr')
+                    ->disk('public')
+                    ->directory('documents')
+                    ->getUploadedFileNameForStorageUsing(function (?Revision $record, TemporaryUploadedFile $file, RelationManager $livewire): string {
+                        /** @var Document */
+                        $document = $livewire->getOwnerRecord();
+
+                        return RevisionResource::getFilename('en', $file->extension(), $document, $record);
+                    }),
+                Forms\Components\FileUpload::make('file.fr')
+                    ->label(__('File (French)'))
+                    ->requiredWithout('file.en')
+                    ->disk('public')
+                    ->directory('documents')
+                    ->getUploadedFileNameForStorageUsing(function (?Revision $record, TemporaryUploadedFile $file, RelationManager $livewire): string {
+                        /** @var Document */
+                        $document = $livewire->getOwnerRecord();
+
+                        return RevisionResource::getFilename('fr', $file->extension(), $document, $record);
+                    }),
+            ]);
+    }
+
+    public function table(Table $table): Table
+    {
+        return $table
+            ->recordTitleAttribute('create_at')
+            ->columns([
+                Tables\Columns\TextColumn::make('created_at')->label(__('Date'))
+                    ->date('Y-m-d'),
+                Tables\Columns\TextColumn::make('has_english')->label(__('English'))
+                    ->badge()
+                    ->color(fn (string $state): string => $state ? 'success' : 'danger')
+                    ->formatStateUsing(fn (bool $state): string => $state ? __('Yes') : __('No')),
+                Tables\Columns\TextColumn::make('has_french')->label(__('French'))
+                    ->badge()
+                    ->color(fn (string $state): string => $state ? 'success' : 'danger')
+                    ->formatStateUsing(fn (bool $state): string => $state ? __('Yes') : __('No')),
+            ])
+            ->headerActions([
+                Tables\Actions\CreateAction::make(),
```

Comment:
Can we add `->createAnother(false)`? At the moment the "Create" button is blocked during the upload process but the "Create and Create Another" one isn't. If you click on this before the upload is completed it will make the revision but not upload the file.

Another option would be to disabled the "Create and create another" button during upload if that's possible.

see: 
- https://filamentphp.com/docs/3.x/actions/prebuilt-actions/create#disabling-create-another
- https://github.com/filamentphp/filament/discussions/6114